google/google.go

```diff
@@ -100,29 +116,33 @@ func GetLoginURL(state string) string {
 	return conf.AuthCodeURL(state)
 }
 
+func WithLoginURL(url string) {
+	loginURL = url
```

Comment:
```suggestion
	loginURL = strings.TrimSpace(s)
```

Reply:
Maybe better to use url.Parse(s), check error and if no error use u.String() to have a clean url string.

Reply:
With the error handling, would you suggest the signature change to `func WithLoginURL(url string) (bool, error)` ?

Reply:
I would say return `error` is enough. Either there is an error or not :)

Reply:
I'd suggest we are aiming for consistency on API. It is either

```
func SetupFromString(redirectURL, clientID string, clientSecret string, scopes []string, secret []byte) error
func WithLoginURL(url string) error
```

or

```
func SetupFromString(redirectURL, clientID string, clientSecret string, scopes []string, secret []byte)
func WithLoginURL(url string)
```

Reply:
I would suggest to go with the first, so return error.

Reply:
Makes sense - that seems more idiomatic too. I'll have that shortly..

Reply:
Sorry for the delay - had some distractions. I was just getting ready to update the PR and noticed that `SetupFromString()` and `Setup()` would differ with regards to their return value. The `Setup()` function could more plausibly have errors, however, errors aren't being returned at this time. I do like the idea of returning an error `WithLoginURL()` since there is possibility for a bad URL being added, but I feel like if I return an error from `SetupFromString()`, I should also return an error for `Setup()`. While that wouldn't break functionality, it could cause linting errors since return values aren't being accepted and handled. I'm going to make the changes for `WithLoginURL()` as prescribed, but I'm interested to get your opinions on handling `SetupFromString()` and `Setup()`.